### PR TITLE
 Allow to open multiple connections via NAT hole punching

### LINKED ISSUE #1834: Prepare NAT hole punching for communication channel
In order to implement direct (p2p) communication channel we need datagram tunnel prepared.

Establish additional datagram tunnel and make it available for p2p communication channel.

### PR DESCRIPTION
**Pull request by soffokl**
Closes #1834

### CODE REVIEW

File: nat/traversal/pinger.go [hunk 1]
```diff
@@ -158,7 +138,58 @@ func (p *Pinger) PingProvider(ip string, localPorts, remotePorts []int, proxyPor
...
+			if err := ipv4.NewConn(res.conn).SetTTL(maxTTL); err != nil {
+				log.Warn().Err(res.err).Msg("Failed to set connection TTL")
+				continue
+			}
+
+			res.conn.Write([]byte("OK")) // notify peer that we are using this connection.
+
+			conns = append(conns, res.conn)
+			if len(conns) == n {
+				return conns, nil
```
Comment (anjmao) [inline nat/traversal/pinger.go]: At this point multiPing can be stopped too.

File: nat/traversal/pinger_test.go [hunk 2]
```diff
@@ -110,6 +102,56 @@ func TestPinger_Provider_Consumer_Ping_Flow(t *testing.T) {
...
+	var pPorts, cPorts []int
+	ports, err := port.NewPool().AcquireMultiple(10)
+	assert.NoError(t, err)
+
+	for i := 0; i < 5; i++ {
+		pPorts = append(pPorts, ports[i].Num())
+		cPorts = append(cPorts, ports[5+i].Num())
+	}
+
+	go consumer.PingPeer("192.168.0.101", cPorts, pPorts, 2, 3)
```
Comment (anjmao) [inline nat/traversal/pinger_test.go]: This ip `192.168.0.101` will probably not work for everyone.

File: nat/traversal/pinger_test.go [hunk 3]
```diff
@@ -110,6 +102,56 @@ func TestPinger_Provider_Consumer_Ping_Flow(t *testing.T) {
...
+	assert.NoError(t, err)
+
+	for i := 0; i < 5; i++ {
+		pPorts = append(pPorts, ports[i].Num())
+		cPorts = append(cPorts, ports[5+i].Num())
+	}
+
+	go consumer.PingPeer("192.168.0.101", cPorts, pPorts, 2, 30)
+	conns, err := provider.PingPeer("127.0.0.1", pPorts, cPorts, 2, 30)
+	assert.EqualError(t, err, "not enough connections")
```
Comment (anjmao) [inline nat/traversal/pinger_test.go]: Test name is Too_Many_Connections but error assertion is `not enough connections`.

File: nat/traversal/pinger.go (hunk 1 above)
Comment (soffokl) [reply nat/traversal/pinger.go]: It will be stopped by the `defer close(stop)`

File: nat/traversal/pinger_test.go (hunk 2 above)
Comment (soffokl) [reply nat/traversal/pinger_test.go]: Wow, thanks. It's a leftover from debugging.

File: nat/traversal/pinger_test.go (hunk 3 above)
Comment (soffokl) [reply nat/traversal/pinger_test.go]: Renamed test

**Pull request merged**